File: CoreEngine/Content/Modules/DevModule/tiles/Stone/AStone.rb
require 'DevModule/tiles/Stone/Stone'

include CoreEngine::Entities
include CoreEngine::Scripting

class AStone < Stone
	def	register
		p "Registered AStone";
		self.unregister_draw_on_layer 2
		set_entity_properties({ 
			draw: true, 
			update: false, 
			click: true 
		})
		@clicked = false
	end

	def on_click(event_bundle)
		@clicked = true
	end

	def draw
		super
		click_color = @clicked ? CoreColor.Black : CoreColor.White
		Canvas.draw_string(CoreFont.System, "A", position.x + 14, position.y + 14, click_color)
	end

end

File: DevModule/Tiles/Astone.cs
using PlumeAPI.Events;
using PlumeAPI.Graphics;
using Microsoft.Xna.Framework;
using PlumeRPG.Entities;

namespace DevModule.Tiles {
	public class AStone : MapTile {

		bool Clicked = false;
		public AStone() : base() {
		}

		public AStone(int x, int y) : base(x, y) {
			UnregisterDrawOnLayer(2);
		} 

		public override void OnClick(EventData bundle) {
			base.OnClick(bundle);
			Clicked = true;
		}

		public override void Draw() {
			base.Draw();
			Canvas.DrawString(FontRepository.System, "A", Position.X + 14, Position.Y + 14, Clicked ? Color.Black : Color.White);
		}
	}
}
